src/geometries/TubeGeometry.js

```diff
@@ -161,16 +166,49 @@ function TubeBufferGeometry( path, tubularSegments, radius, radialSegments, clos
 			normal.z = ( cos * N.z + sin * B.z );
 			normal.normalize();
 
-			normals.push( normal.x, normal.y, normal.z );
-
 			// vertex
 
-			vertex.x = P.x + radius * normal.x;
-			vertex.y = P.y + radius * normal.y;
-			vertex.z = P.z + radius * normal.z;
+			var r = radius instanceof Path ? radius.getPointAt( i / tubularSegments ).y : radius;
+			vertex.x = P.x + r * normal.x;
+			vertex.y = P.y + r * normal.y;
+			vertex.z = P.z + r * normal.z;
 
 			vertices.push( vertex.x, vertex.y, vertex.z );
 
+			// adjust normals if the radius is a Path
+
+			if ( radius instanceof Path ) {
+
+				var delta = 0.0001;
+				var t1 = i / tubularSegments - delta;
+				var t2 = i / tubularSegments + delta;
+
+				// Capping in case of danger
+
+				if ( t1 < 0 ) t1 = 0;
+				if ( t2 > 1 ) t2 = 1;
+
+				path.getPointAt( t1, P1 );
```

Comment:
Isn't the 2nd argument still optional? That means the user may assume a new vector will be created. That is why I wrote it as:
```javascript
P1 = path.getPointAt( t1, P1 );
```
/ping @Mugen87

Reply:
I think that not all classes extending Curve (https://github.com/mrdoob/three.js/blob/master/src/extras/core/Curve.js#L63) honor that signature. (https://github.com/mrdoob/three.js/blob/master/src/extras/core/CurvePath.js#L58).  I've fixed the code so it doesn't pass any optional argument and always expects the returned value. It would be a good optimisation to take the optional vec3 down through all the calls.

Reply:
You did the opposite of what I expected.  Now, if the user-defined `getPoint()` method expects a `target`, the code will not work...

https://github.com/mrdoob/three.js/issues/12231

Reply:
It's quite confusing, to be honest. Reverted.

Reply:
Yes. But at some point we remove the optional targets from the curve classes. Things should be more consistent then.